src/core/basetypes/MCBase64.c

```diff
@@ -140,23 +140,56 @@ char * MCDecodeBase64ByLines(const char * in, int len, int * p_outlen) {
         }
 
         c1 = in[0];
-        c2 = in[1];
-        c3 = in[2];
-        c4 = in[3];
-        if (CHAR64(c1) == -1 || CHAR64(c2) == -1 ||
-            (c3 != '=' && CHAR64(c3) == -1) ||
-            (c4 != '=' && CHAR64(c4) == -1)) {
+        c2 = 0;
+        c3 = 0;
+        c4 = 0;
+
+        while (in < end) {
+            in++;
```

Comment:
It seems a char at in==2 is skipped.
Also, it seems we can out of bound here.

Reply:
No. c1 is in[0], then we going inside this loop, then in++.